programs/dynamic-bonding-curve/src/instructions/migration/meteora_damm/meteora_damm_lock_lp_token.rs

```diff
@@ -119,13 +126,179 @@ pub fn handle_migrate_meteora_damm_lock_lp_token<'info>(
+            lp_exclude_fee
+        }
+        (false, true) => {
+            let lp_to_lock = migration_metadata.validate_and_get_creator_lock_amount()?;
+            let lp_exclude_fee = exclude_fee_lp_amount(lp_to_lock, damm_migration_accounts)?;
+            migration_metadata.lock_as_creator(lp_exclude_fee);
+            lp_exclude_fee
+        }
         (false, false) => return Err(PoolError::InvalidOwnerAccount.into()),
     };
 
     ctx.accounts
         .lock(const_pda::pool_authority::BUMP, lp_to_lock)
 }
+
+struct DammAccounts<'c, 'info> {
+    pool: &'c AccountInfo<'info>,
+    lp_mint: &'c AccountInfo<'info>,
+    a_vault: &'c AccountInfo<'info>,
+    b_vault: &'c AccountInfo<'info>,
+    a_vault_lp: &'c AccountInfo<'info>,
+    b_vault_lp: &'c AccountInfo<'info>,
+    a_vault_lp_mint: &'c AccountInfo<'info>,
+    b_vault_lp_mint: &'c AccountInfo<'info>,
+}
+
+#[inline(never)]
+fn validate_damm_accounts(accounts: &DammAccounts<'_, '_>) -> Result<()> {
+    let DammAccounts {
+        pool,
+        lp_mint,
+        a_vault,
+        b_vault,
+        a_vault_lp,
+        b_vault_lp,
+        a_vault_lp_mint,
+        b_vault_lp_mint,
+    } = accounts;
+
+    require!(
+        pool.owner.eq(&dynamic_amm::ID),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    let pool_state = Pool::try_deserialize(&mut pool.data.borrow_mut().as_ref())?;
+
+    require!(
+        pool_state.lp_mint.eq(lp_mint.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    require!(
+        pool_state.a_vault.eq(a_vault.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    require!(
+        pool_state.b_vault.eq(b_vault.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    require!(
+        pool_state.a_vault_lp.eq(a_vault_lp.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    require!(
+        pool_state.b_vault_lp.eq(b_vault_lp.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    let expected_a_vault_lp_mint = token::accessor::mint(a_vault_lp)?;
+
+    require!(
+        expected_a_vault_lp_mint.eq(a_vault_lp_mint.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    let expected_b_vault_lp_mint = token::accessor::mint(b_vault_lp)?;
+
+    require!(
+        expected_b_vault_lp_mint.eq(b_vault_lp_mint.key),
+        PoolError::InvalidMigrationAccounts
+    );
+
+    Ok(())
+}
+
+fn get_damm_virtual_price(accounts: &DammAccounts<'_, '_>) -> Result<u128> {
+    validate_damm_accounts(accounts)?;
+
+    let DammAccounts {
+        lp_mint,
+        a_vault,
+        b_vault,
+        a_vault_lp,
+        b_vault_lp,
+        a_vault_lp_mint,
+        b_vault_lp_mint,
+        ..
+    } = accounts;
+
+    let current_time: u64 = Clock::get()?
+        .unix_timestamp
+        .try_into()
+        .map_err(|_| PoolError::MathOverflow)?;
+
+    let a_vault = Vault::try_deserialize(&mut a_vault.data.borrow_mut().as_ref())?;
```

Comment:
Why not use Box<Account<'info, Vault>> in the context, so we can skip all `try_deserialize `

Reply:
Fixed